### ISSUE TITLE
Issue with handling InvalidJwtAuthenticationException

### ISSUE THREAD

**Issue by sen4ik** [hantsy/spring-webmvc-jwt-sample#7]
Hi,

First of all thank you for your project. It is helpful.

Second, for some reason, RestExceptionHandler handles VehicleNotFoundException without any issues, but is not handling InvalidJwtAuthenticationException. Do you have any idea why?

Can it be because InvalidJwtAuthenticationException exception is thrown from JwtTokenProvider which is not a controller?

**Comment by sen4ik** [comment]
The only way i could figure out is create another filter. What would you say about this approach?

`@Component
@Order(1)
public class CustomFilter implements Filter {

    @Override
    public void doFilter(
            ServletRequest request,
            ServletResponse response,
            FilterChain chain) throws IOException {

        HttpServletResponse res = (HttpServletResponse) response;
        try {
            chain.doFilter(request, response);
        }
        catch(Exception e){
            if(e instanceof InvalidJwtAuthenticationException){
                int code = 400;
                JSONObject entity = new JSONObject();
                entity.put("status", code);
                entity.put("message", e.getMessage());
                res.setContentType(MediaType.APPLICATION_JSON_VALUE);
                res.setStatus(code);
                res.getWriter().write(entity.toString());
            }
        }
    }
}`

**Comment by hantsy** [comment]
Just verifed this issue,  I maybe have not tested it at all.
There are some discussions on google results.

https://stackoverflow.com/questions/19767267/handle-spring-security-authentication-exceptions-with-exceptionhandler

**Comment by hantsy** [comment]
`InvalidJwtAuthenticationException` is a little special, it is thrown in Spring Security handling progress.

**Comment by sen4ik** [comment]
Will you have time to implement it for your project?

**Comment by hantsy** [comment]
@sen4ik If your approach is working well, why not keep with it. Ideally, I prefer standard OAuth2/Oidc flow in these days.

**Comment by hantsy** [comment]
Currently, I am focusing on other personal projects, maybe I will go through this problem and resolve it. Let it open until it is resolved.  Thanks for reporting this issue @sen4ik .

**Comment by gastonestol** [comment]
Any solution yet?

I did this because I couldn't make it pass through JwtAuthenticationEntryPoint exception handling

`public class JwtTokenFilter extends GenericFilterBean {

    private final JwtTokenProvider jwtTokenProvider;

    public JwtTokenFilter(JwtTokenProvider jwtTokenProvider) {
        this.jwtTokenProvider = jwtTokenProvider;
    }

    @Override
    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
        try {
            String token = jwtTokenProvider.resolveToken((HttpServletRequest) request);
            if (token != null && jwtTokenProvider.validateToken(token)) {
                Authentication auth = jwtTokenProvider.getAuthentication(token);
                SecurityContextHolder.getContext().setAuthentication(auth);
            }
            chain.doFilter(request, response);
        } catch(InvalidJwtAuthenticationException e){
            HttpServletResponse res = (HttpServletResponse) response;
            res.setStatus(HttpServletResponse.SC_CONFLICT);
        }

    }
}`

**Comment by sen4ik** [comment]
@gastonestol Hi. There is a project I have where I used JWT token filter from hantsy and did some fixes around it. You can see it here https://github.com/sen4ik/vfb.

**Comment by hantsy** [comment]
@sen4ik  In the latest codes, I used the built-in `HttpStatusEntryPoint` instead of my custom EntryPoint. 

https://github.com/hantsy/spring-webmvc-jwt-sample/blob/master/src/main/java/com/example/demo/config/SecurityConfig.java#L35

And I removed `InvalidJwtAuthenticationException`.

If you want to control the exception in detail, it is better to create a series of components(Handlers for failure and uccess) for your EntryPoint. eg.
